File: Homework_2/earth_electrons.cpp
// Copyright 2017 Michael Graziano mjgrazia@bu.edu

#include <iostream>
#include <cstdint>
#include <cfloat>
#include <cmath>

using namespace std;

int main(){
    double earth_mass = 5.972 * pow(10, 24);
    double atoms_in_mole = 6.022 * pow(10, 23);
    double tera_convert = 4 * pow(10, 12);
    double element_info[9][3] = {
        {0.321, 55.84 * pow(10, -3), 26},  // Iron Values
        {0.301, 15.99 * pow(10, -3), 8},   // Oxygen Values
        {0.151, 28.08 * pow(10, -3), 14},  // Silicon Values
        {0.139, 24.30 * pow(10, -3), 12},  // Magnesium Values
        {0.029, 32.06 * pow(10, -3), 16},  // Sulfur Values
        {0.018, 58.69 * pow(10, -3), 28},  // Nickel Values
        {0.015, 40.07 * pow(10, -3), 20},  // Calcuim Values
        {0.014, 26.998 * pow(10, -3), 13}, // Aluminum Values 
        {0.012, 75 * pow(10, -3), 60}      // Other Values
    };
    double electrons = 0;
    double terabytes;

for (int e=0; e<9; e++){
    electrons = electrons + earth_mass * element_info[e][0] / element_info[e][1] * atoms_in_mole * element_info[e][2];
}

terabytes = electrons / tera_convert;
cout << terabytes << endl << terabytes - pow(10, 38) << endl << terabytes + pow(10, 38) << endl;

return 0;
}

File: Homework_2/earth_electrons.py
# Copyright 2017 Michael Graziano mjgrazia@bu.edu

import sys
import numpy

earth_mass = 5.972 * (10**24)
atoms_in_mole = 6.022 * (10**23)
tera_convert = 4 * (10**12)
element_info = {"Iron": [0.321, 55.84 * (10**-3), 26],
                "Oxygen": [0.301, 15.99 * (10**-3), 8],
                "Silicon": [0.151, 28.08 * (10**-3), 14],
                "Magnesium": [0.139, 24.30 * (10**-3), 12],
                "Sulfur": [0.029, 32.06 * (10**-3), 16],
                "Nickel": [0.018, 58.69 * (10**-3), 28],
                "Calcuim": [0.015, 40.07 * (10**-3), 20],
                "Aluminum": [0.014, 26.998 * (10**-3), 13],
                "Other": [0.012, 75 * (10**-3), 60]
               }

electrons = numpy.float32(0)
for element in element_info:
   electrons = (electrons
               + earth_mass 
               * element_info[element][0] 
               / element_info[element][1] 
               * atoms_in_mole 
               * element_info[element][2]
               )
terabytes = electrons / tera_convert 

print(terabytes, 
      terabytes - 1*(10**38), 
      terabytes + 1*(10**38),
      sep = "\n"
     )

